### ISSUE TITLE
Issue with search of a node with his DNS CNAME

### ISSUE THREAD

**Issue by earendilfr** [netdisco/netdisco#1213]
## Expected Behavior
I have an issue with the node search.
When I search a node with a DNS name, I receive the following error message:
![image](https://github.com/netdisco/netdisco/assets/2353530/ecfecf8b-d34e-47e2-b893-15869c31962e)

And, in netdisco-web logs, I can see this error message:
```
[4138832] 2024-05-24 12:18:12 error request to GET /ajax/content/search/node crashed: core - template - undef error - DBIx::Class::Row::get_column(): No such column 'router_ip' on App::Netdisco::DB::Result::NodeIp at /opt/netdisco/perl5/lib/perl5/App/Netdisco/DB/Result/NodeIp.pm line 270
```

## Steps to Reproduce (for bugs)

1. Search the DNS CNAME entry of a device in node parts
    By example, I have here the CNAME `cname-switch-1` related to the device `switch-1` and I search the node `cname-switch-1`
2. I see this error message in web page

## Your Environment

* Netdisco version used: 2.76.0
* SNMP::Info version used: 3.970001
* DB Schema: 87

## Config info (deployment.yml)

```yaml
#
# NETDISCO 2 CONFIGURATION FILE
#
# Settings in this file override share/config.yml
# See App::Netdisco::Manual::Configuration for more info.

# ------------------
# ESSENTIAL SETTINGS
# ------------------

database:
  name: 'netdiscodb'
  user: 'netdisco'
  pass: 'xxxxxxxx'

# -----------------
# LOG SETTINGS
# -----------------
log: 'warn'
log_format: '[%T] {PID: %P} {Level:%L} %m'

# set to false if you MUST maintain backwards compatibility
# with Netdisco 1.x web frontend.
# `````````````````````````````````````````````````````````
safe_password_store: true

# SNMP community string(s)
# ````````````````````````
device_auth:
  xxxxxxxx

# this is the schedule for automatically keeping netdisco up-to-date
# ``````````````````````````````````````````````````````````````````
schedule:
  discoverall:
    when: 
      min: 0
      hour: '11,23'
  macwalk:
    when:
      min: '0,30'
  arpwalk:
    when:
      min: '15,45'
  nbtwalk: null
  expire:
    when: 
      min: 20
      hour: 23
  makerancidconf:
    when: 
      min: 30
      hour: '5,17'
  makeansiblehosts: null

# do not discover IP Phones or Wireless Access Points.
# usually these are visible as device neighbors but don't support
# SNMP, which just clogs up the job queue.
# ```````````````````````````````````````````````````````````````
#discover_no_type:
#  - '(?i)phone'
#  - '(?i)cisco\s+AIR-(L|C)AP'
discover_waps: false
discover_phones: false

# number of SNMP workers to run in parallel (in netdisco-backend).
# the default is twice the number of CPU cores. increase this if
# your system has few cores and the schedule is taking too long.
# ```````````````````````````````````````````````````````````````
workers:
  tasks: 'AUTO * 4'
  timeout: 600
  sleep_time: 1
  min_runtime: 0
  max_deferrals: 10
  retry_after: '1 day'

# number of parallel DNS queries for node names
# `````````````````````````````````````````````
dns:
  max_outstanding: 50

# Domain suffix to remove
# `````````````````````````````````````````````
domain_suffix:
  - 'xxxxxxxx'

# Configure the interface used for device identity
# ````````````````````````````````````````````````
device_identity:
  xxxxxxxx

# We disable completely the Netbios discover
# `````````````````````````````````````````````````````````````````````
nbtstat_no:
  - 'any'

# Global tuning for MAC suck to take in count users in VLAN
# WARNING: on Cisco switch, you should allow polling on SNMP context for each VLAN
# `````````````````````````````````````````````````````````````````````
macsuck_all_vlans: true

# We clear job that didn't have run after 2 days
# `````````````````````````````````````````````````````````````````````
expire_jobs: 2

# Turn this on to have Netdisco do a reverse lookup of the device sysName.0 field 
# to use as the management IP address for a device.
# `````````````````````````````````````````````````````````````````````
reverse_sysname: true

# Web tune parameters
# `````````````````````````````````````````````````````````````````````
table_pagesize: 50


# Custom reports
# `````````````````````````````````````````````````````````````````````
reports:
  xxxxxxxx

# LDAP authentication
# `````````````````````````````````````````````````````````````````````
ldap:
  xxxxxxxx

# vim: tabstop=2 expandtab
```

**Comment by ollyg** [comment]
Hi @earendilfr I suspect I know what the cause of this bug is, sorry about it

**Comment by ollyg** [comment]
this should be fixed in 365601f2 but let me know if not

**Comment by ollyg** [comment]
super, many thanks!
